src/Common/test/Common.Test/Configuration/StringExtensionsTest.cs

```diff
@@ -0,0 +1,29 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using FluentAssertions;
+using Steeltoe.Common.Configuration;
+using Xunit;
+
+namespace Steeltoe.Common.Test.Configuration;
+
+public sealed class StringExtensionsTest
```

Comment:
It's unclear to me what escaping rules come into play. I've googled a bit and found the following:
- yaml keys that contain a dot or whitespace (and several other characters) need to be surrounded by brackets
	https://stackoverflow.com/questions/34070987/escaping-a-dot-in-a-map-key-in-yaml-in-spring-boot
	https://docs.spring.io/spring-boot/docs/2.2.1.RELEASE/reference/html/spring-boot-features.html#boot-features-external-config-relaxed-binding
	https://artofcode.wordpress.com/2020/10/29/special-characters-in-a-key-in-spring-boot-yaml-file/
	https://stackoverflow.com/questions/57795723/sping-boot-config-how-to-keep-whitespace-in-yaml-key-being-used-to-populate-map
	https://stackoverflow.com/questions/65346054/ignore-specify-slash-character-in-yaml-config-for-spring-boot-admin-application
- special handling of \$
	https://docs.spring.io/spring-cloud-config/reference/server/environment-repository/property-overrides.html
- unicode escapes and backslashes
	https://www.w3schools.io/file/properties-escape-characters/
	https://en.wikipedia.org/wiki/.properties
	https://www.kajabity.com/kajabity-tools/java-properties-classes/the-java-properties-file-format/

It would help to point to some spec references (in tests or in the implementation), so we'll know what needs to be accounted for,

Reply:
I am also unclear on what should apply. Escaping was not a topic I was intending to address in any way other than preserving what was here before